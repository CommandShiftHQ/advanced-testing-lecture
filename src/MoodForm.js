import React from "react";

const MoodForm = ({ isMoodCaptured, handleSubmit, handleInputChange }) => {
  if (isMoodCaptured) {
    return <div>Thanks for sharing your mood.</div>;
  }

  return (
    <form className="search-form" onSubmit={handleSubmit} data-testid="form">
      <input
        className="search-input"
        type="text"
        placeholder="how are you feeling today?"
        size="200"
        onChange={handleInputChange}
      />
      <br />
      <button name="go" className="submit-button" type="submit">
        submit
      </button>
    </form>
  );
};

export default MoodForm;
