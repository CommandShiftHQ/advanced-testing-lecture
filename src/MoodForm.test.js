import React from "react";
import { render, fireEvent } from "@testing-library/react";
import MoodForm from "./MoodForm";

test("it renders", () => {
  const mockHandleInputChange = jest.fn();
  const mockHandleSubmit = jest.fn();

  const { asFragment } = render(
    <MoodForm
      isMoodCaptured={false}
      handleInputChange={mockHandleInputChange}
      handleSubmit={mockHandleSubmit}
    />
  );

  expect(asFragment()).toMatchSnapshot();

  /*

  const textbox = getByRole("textbox");
  const button = getByRole("button", { name: /submit/i });

  fireEvent.change(textbox, {
    target: { value: "feeling great!" },
  });
  // fireEvent.click(button);
  expect(textbox.value).toBe("feeling great!");
  await wait(() => {
    expect(saveMood).toHaveBeenCalledTimes(1);
    expect(saveMood).toHaveBeenCalledWith({
      mood: "feeling great!",
    });
  });
  */
});

test("it renders success when user submitted the form", () => {
  const mockHandleInputChange = jest.fn();
  const mockHandleSubmit = jest.fn();

  const { asFragment } = render(
    <MoodForm
      isMoodCaptured={true}
      handleInputChange={mockHandleInputChange}
      handleSubmit={mockHandleSubmit}
    />
  );

  expect(asFragment()).toMatchSnapshot();
});

test("it submits the form data and call handle submit function", () => {
  const mockHandleInputChange = jest.fn();
  const mockHandleSubmit = jest.fn();

  const {  getByTestId } = render(
    <MoodForm
      isMoodCaptured={false}
      handleInputChange={mockHandleInputChange}
      handleSubmit={mockHandleSubmit}
    />
  );

  const form = getByTestId("form");
  fireEvent.submit(form);
  expect(mockHandleSubmit).toHaveBeenCalled();
});
