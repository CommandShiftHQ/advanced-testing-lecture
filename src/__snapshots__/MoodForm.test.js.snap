// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<DocumentFragment>
  <form
    class="search-form"
    data-testid="form"
  >
    <input
      class="search-input"
      placeholder="how are you feeling today?"
      size="200"
      type="text"
    />
    <br />
    <button
      class="submit-button"
      name="go"
      type="submit"
    >
      submit
    </button>
  </form>
</DocumentFragment>
`;

exports[`it renders success when user submitted the form 1`] = `
<DocumentFragment>
  <div>
    Thanks for sharing your mood.
  </div>
</DocumentFragment>
`;
